koku/api/iam/serializers.py

```diff
@@ -131,11 +134,14 @@ def create(self, validated_data):
 class UserPreferenceSerializer(serializers.ModelSerializer):
     """Serializer for the UserPreference model."""
 
+    user_uuid = serializers.StringRelatedField(source='user.uuid')
+
     class Meta:
         """Metadata for the serializer."""
 
         model = UserPreference
-        fields = ('__all__')
+        fields = ('uuid', 'name', 'description', 'preference', 'user', 'user_uuid')
+        extra_kwargs = {'user': {'write_only': True, 'required': False}}
```

Comment:
Have you looked at this in the Browesable API with `write_only: True` I'm betting it has a selection for user.

I'd guess you could accomplish what you are looking for  with `read_only: True` and without the use of the `user_uuid` field. You could look at the following for what was done with Providers to establish the `created_by`:
https://github.com/project-koku/koku/blob/master/koku/api/provider/serializers.py#L76

Reply:
The problem I was trying to solve was that `create()` needed to be able to set the `user_id`, but I wanted `get()` to display something more meaningful than the internal db `user_id`. I couldn't remember how we had solved that for `Customer`, so this was my lame kludge solution.